analyzers/tests/SonarAnalyzer.UnitTest/Common/ReadMeTest.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using System.Text.RegularExpressions;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Utilities;
+
+namespace SonarAnalyzer.UnitTest.Common;
+
+[TestClass]
+public class ReadMeTest
+{
+    public TestContext TestContext { get; set; }
+
+    private string ReadMe { get; set; }
+
+    [TestInitialize]
+    public void Init()
+    {
+        var readme = new FileInfo(Path.Combine(TestContext.DeploymentDirectory, "./../../../../../../README.md"));
+        using var reader = readme.OpenText();
+        ReadMe = reader.ReadToEnd();
```

Comment:
Simplier APIs and Windows path separator 
```suggestion
        var path = Path.Combine(TestContext.DeploymentDirectory, ".\..\..\..\..\..\..\README.md");
        return File.ReadAllText(path);
```
I'd be also fine with
```
public void Init() =>
    readMe = Path.ReadAllText(Path.Combine(TestContext.DeploymentDirectory, ".\..\..\..\..\..\..\README.md"));
```

Reply:
I went for the last one. I chose `'/'` over `'\'` originally because in my experience, Windows systems deal equally good with both, but UNIX based systems fail to deal correctly they other way around. (And it requires an extra `@` before the string literal)